xla/service/gpu/gpu_windowed_einsum_handler.cc

```diff
@@ -638,12 +637,15 @@ class WindowedEinsumVisitor : public DfsHloRewriteVisitor {
   absl::Status HandleDot(HloInstruction* dot) override {
     CHECK_EQ(dot->opcode(), HloOpcode::kDot);
     HloComputation* comp = dot->parent();
-    // Now that we have processed all loops, we can check if there are any
-    // allgather-dot pattern that we can optimize. We'd want to transform:
+    // Rewrites an allgather-dot pattern that shares the same operand with a
+    // windowed einsum loop to consume the output of the loop and remove the
+    // all-gather. Now that we have processed all loops, we can check if there
+    // are any allgather-dot pattern that we can optimize. We'd want to
+    // transform:
     //                       input
     //                       /    |
-    //                dequantize  |
-    //                  (FP8)     |
+    //               dequantize   |
+    //          (optional if FP8) |
```

Comment:
`(optional if FP8)` implies, to me at least, that the dequantize is only optional if FP8 (and otherwise required).

Change to `(optional, only if FP8)` or `(optional)` or `(only if FP8)`